### PR TITLE
Add download_yelt and use auto_retry

### PR DESCRIPTION
**Pull request by DanielArndt**
- I only wrote tests for one of each "kind" of metrics request for auto_retry -- but I'm open to doing it for the rest if someone thinks it would be useful.

### CODE REVIEW

File: tests/test_base_resources.py [hunk 1]
```diff
@@ -441,6 +442,28 @@ def test_ep(self, reqmock):
         assert tm[0].num == 1.0
         assert tm[1].num == 2.0
 
+    def test_ep_auto_retry_true(self, reqmock):
+        responses = [
+            {'status_code': 503, 'headers': {'Retry-After': '0.01'}},
```
Comment (pmdarrow) [inline tests/test_base_resources.py]: It's probably not a big deal that this sleeps for .01s, but I'd probably mock it like we do here: https://github.com/analyzere/analyzere-python/blob/080baaad3d7cd3a43894a9abb88f625fa959ab01/tests/test_requestor.py#L140. Then you could assert it was called.

**Comment by pmdarrow** [comment]
Just one comment, otherwise LGTM.

File: tests/test_base_resources.py (hunk 1 above)
Comment (DanielArndt) [inline tests/test_base_resources.py]: ah nice! good call

Comment (DanielArndt) [inline tests/test_base_resources.py]: done in https://github.com/analyzere/analyzere-python/pull/15/commits/9d798676c87158b0b21571ccef9b31c9c4fbff83

**Pull request merged**